### PR TITLE
Add drone ci registration

### PR DESCRIPTION
**Pull request by mariusvniekerk**
This makes use of the drone-cli go binary.

Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

File: conda_smithy/ci_register.py [hunk 1]
```diff
@@ -35,6 +36,17 @@
...
     )
 
+try:
+    with open(os.path.expanduser("~/.conda-smithy/drone.token"), "r") as fh:
+        drone_token = fh.read().strip()
+    if not drone_token:
+        raise ValueError()
+except (IOError, ValueError):
+    print(
+        "No appveyor token. Create a token at https://cloud.drone.io/account and\n"
```
Comment (isuruf) [inline conda_smithy/ci_register.py]:
```suggestion
        "No drone token. Create a token at https://cloud.drone.io/account and\n"
```

**Comment by jjhelmus** [comment]
Another option would be to use the API, https://docs.drone.io/api/repos/repo_create/

**Comment by scopatz** [comment]
LGTM to get started, but news / tests would be good add.

**Comment by jakirkham** [comment]
> This makes use of the drone-cli go binary.

Should we add it to the requirements?

**Comment by pkgw** [comment]
Xref: conda-forge/conda-smithy#1079

**Comment by scopatz** [comment]
There are now conflicts here

**Comment by mariusvniekerk** [comment]
Seems like drone updated their docs to point to here https://docs.drone.io/api/endpoints/repos/repo_create/

**Comment by mariusvniekerk** [comment]
Companion in staged recipes: https://github.com/conda-forge/staged-recipes/pull/9511

**Comment by isuruf** [comment]
Can you add a news entry?

**Comment by mariusvniekerk** [comment]
@isuruf, manually invoked the methods and used them to register the rust feedstock.

**Comment by isuruf** [comment]
Can you change the default method for `aarch64` to `drone` ?

**Comment by mariusvniekerk** [comment]
Does drone also do armv7?

**Comment by jjhelmus** [comment]
> Does drone also do armv7?

Drone provides a 32-bit ARM environment but I believe uname reports armv8l not armv7l.

**Comment by isuruf** [comment]
Thanks

**Pull request merged**